### ISSUE TITLE
Pressing enter in page title ignores page level template lock and creates paragraph block.

### ISSUE THREAD

**Issue by james-Ballyhoo** [WordPress/gutenberg#14568]
**Describe the bug**
When the enter key is pressed while editing the page title, the editor creates a paragraph at the top of the content, ignoring the post-type level template lock. This paragraph can't be deleted and invalidates the content when next loaded.

**To Reproduce**
1. Setup a template + template_lock for a post type as per https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-templates/#locking
```php
<?php
function myplugin_register_template() {
    $post_type_object = get_post_type_object( 'post' );
    $post_type_object->template = array(
        array( 'core/gallery'),
    );
    $post_type_object->template_lock = 'all';
}
add_action( 'init', 'myplugin_register_template' );
```
2. Go to the post editor
3. type into the title, and press enter
4. A paragraph gets added before the gallery.

**Expected behavior**
Should do either nothing, or focus into the first block in the content.

**Additional context**
- Wordpress version: 5.1.1

**Comment by youknowriad** [comment]
I think this is probably a duplicate issue and @jorgefilipecosta was working on it.

**Comment by jorgefilipecosta** [comment]
PR https://github.com/WordPress/gutenberg/pull/14003 addresses this problem.

**Comment by youknowriad** [comment]
this looks like a duplicate of #10186

**Comment by FerrielMelarpis** [comment]
@jorgefilipecosta sorry for the tag but I am still experiencing this issue on the gutenberg version on wordpress 5.1.1
![bug](https://user-images.githubusercontent.com/8219804/58678908-f7a92780-8393-11e9-99a8-fb760499c6a0.gif)

**Comment by aduth** [comment]
@FerrielMelarpis Can you try updating to the latest version of WordPress (5.2.1) and test again? The fix would not have been available as of WordPress 5.1.1.

**Comment by FerrielMelarpis** [comment]
Thanks @aduth I'll try as soon as Pagely bumped our version.
Might be a noob question but, how do I know the Gutenberg tag/version/branch I should check out to match what Wordpress is using?
I can only see the wp-5.1-beta tag. Is Wordpress's Gutenberg in sync with `master` branch?

**Comment by aduth** [comment]
There was a similar discussion in Slack yesterday ([link requires registration](https://make.wordpress.org/chat/)):

https://wordpress.slack.com/archives/C02QB2JS7/p1559591232147400

tl;dr is that the plugin / repository does not always map cleanly to a WordPress version. WordPress 5.2 is roughly equivalent to Gutenberg 5.3 (but also includes bug fixes from later plugin releases).
